src/core/ipfs/files.js

```diff
@@ -53,6 +53,7 @@ module.exports = function files (self) {
       pull(
         pull.values([hash]),
         pull.asyncMap(self._dagS.get.bind(self._dagS)),
+        pull.take(1),
```

Comment:
no it doesn't add memory overhead, because the implementation of `.get` in the dagService is literally the same thing